tests/Unit/Domain/Test_Neighbors.cpp

```diff
@@ -0,0 +1,80 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/Neighbors.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+TEST_CASE("Unit.Domain.Neighbors", "[Domain][Unit]") {
+  // Test default constructor:
+  Neighbors<3> test_neighbors;
+
+  // Test constructor:
+  const Orientation<3> custom_orientation(std::array<Direction<3>, 3>{
+      {Direction<3>::upper_eta(), Direction<3>::upper_zeta(),
+       Direction<3>::upper_xi()}});
+
+  std::unordered_set<ElementId<3>> custom_ids;
+  std::array<SegmentId, 3> segment1_ids
+      {{SegmentId(2, 3), SegmentId(1, 0), SegmentId(1, 1)}};
+  ElementId<3> element1_id(2, segment1_ids);
+  custom_ids.insert(element1_id);
+
+  std::array<SegmentId, 3> segment2_ids
+      {{SegmentId(2, 2), SegmentId(1, 1), SegmentId(1, 0)}};
+  ElementId<3> element2_id(3, segment2_ids);
+  custom_ids.insert(element2_id);
+
+  std::array<SegmentId, 3> segment3_ids
+      {{SegmentId(2, 1), SegmentId(1, 0), SegmentId(1, 1)}};
+  ElementId<3> element3_id(1, segment3_ids);
+  custom_ids.insert(element3_id);
+
+  Neighbors<3> custom_neighbors(custom_ids, custom_orientation);
+
+  // Test size
+  CHECK(custom_neighbors.size() == 3);
+
+  // Test output
+  // This test is removed until we have proper output for unordered_sets
+  /*CHECK(get_output(custom_neighbors) ==
+        "Ids = "
+        "([B1,(L2I1,L1I0,L1I1)],[B3,(L2I2,L1I1,L1I0)],[B2,(L2I3,L1I0,L1I1)]); "
+        "orientation = (+1, +2, +0)");*/
+
+  // Test add_ids
+  std::unordered_set<ElementId<3>> more_custom_ids;
+  std::array<SegmentId, 3> segment4_ids{
+      {SegmentId(2, 3), SegmentId(1, 0), SegmentId(1, 1)}};
+  ElementId<3> element4_id(0, segment4_ids);
+  more_custom_ids.insert(element4_id);
+
+  custom_neighbors.add_ids(more_custom_ids);
+  CHECK(custom_neighbors.size() == 4);
+  /*CHECK(get_output(custom_neighbors) ==
+        "Ids = "
+        "([B1,(L2I1,L1I0,L1I1)],[B3,(L2I2,L1I1,L1I0)],[B0,(L2I3,L1I0,L1I1)],["
+        "B2,(L2I3,L1I0,L1I1)]); "
+        "orientation = (+1, +2, +0)");*/
+  // Test set_ids_to:
+  custom_neighbors.set_ids_to(custom_ids);
+  CHECK(custom_neighbors.size() == 3);
+  /*CHECK(get_output(custom_neighbors) ==
+        "Ids = "
+        "([B1,(L2I1,L1I0,L1I1)],[B3,(L2I2,L1I1,L1I0)],[B2,(L2I3,L1I0,L1I1)]); "
+        "orientation = (+1, +2, +0)");*/
+
+  // Test serialization:
+  CHECK(custom_neighbors == (serialize_and_deserialize(custom_neighbors)));
+
+  // Test comparison:
+  CHECK(test_neighbors == test_neighbors);
+  CHECK(custom_neighbors == custom_neighbors);
+  CHECK(test_neighbors != custom_neighbors);
+
+  // Test semantics:
+  auto custom_copy = custom_neighbors;
+  test_copy_semantics(test_neighbors);
+  test_move_semantics(custom_neighbors, custom_copy);
```

Comment:
Is this a new change? Because my Travis builds are failing and complaining about this even though it builds on my local Wheeler (I haven't rebased)